packages/core/strapi/lib/commands/transfer/utils.js

```diff
@@ -242,10 +242,26 @@ const loadersFactory = (defaultLoaders = {}) => {
   };
 };
 
+/**
+ * Get the telemetry data to be sent for a didDEITSProcess* event from an initialized transfer engine object
+ *
+ * @param {import('@strapi/data-transfer/types').ITransferEngine} engine Initialized transfer engine
+ * @returns {object} Telemetry properties object
+ */
+const getTransferTelemetryPayload = (engine) => {
+  return {
+    eventProperties: {
+      source: engine?.sourceProvider?.name,
```

Comment:
Should we add a default value in case this value is undefined? (eg: 'N/A' or 'Empty')

Reply:
Ping @marcoautiero

Reply:
Do we have cases in which this value is undefined? 👀

Reply:
Only if we called this function with an uninitialized engine. In import/export/transfer commands, we exit the process before without sending any events if that happens (because we only send transfer error events if the transfer actually tried to start)

Reply:
undefined is fine to me